### PR TITLE
Feature: Improved commands for Pin/Unpin

### PR DESCRIPTION
**Pull request by cinqmilleans**
**Resolved / Related Issues**
Improves commands Pin/UnPin (text, perf, IsExecutable criteria more complete, IsExecutable notify even from FileExplorer).
Fix a modified list exception during its enumeration (when IsExecutable is updated while the favorites list is modified).

- [ ] Were these changes approved in an issue or discussion with the project maintainers? In order to prevent extra work, feature requests and changes to the codebase must be approved before the pull request will be reviewed. This prevents extra work for the contributors and maintainers.
   Closes #issue...

**Validation**
How did you test these changes?
- [x] Did you build the app and test your changes?
- [ ] Did you check for accessibility? You can use Accessibility Insights for this.
- [x] Did you remove any strings from the en-us resource file?
   - [x] Did you search the solution to see if the string is still being used? 
- [ ] Did you implement any design changes to an existing feature?
   - [ ] Was this change approved?
- [ ] Are there any other steps that were used to validate these changes?

### CODE REVIEW

**Comment by yaira2** [comment]
> Are there any other steps that were used to validate these changes?

Can you fill out this section? It helps me know what to test.

**Comment by cinqmilleans** [comment]
There is no visible change for the user, just more complete code (for example if we add a button, its state will always be correct, it was not the case before). There are also hard-to-predict exceptions that have been fixed.

**Comment by yaira2** [comment]
Is there a good way to test the state?

**Comment by cinqmilleans** [comment]
Edit bookmarks in FileExplorer, and see if this is reflected in the context menu. The state is no longer checked when the menu is opened, it is an updated value when it detects a modification.

File: src/Files.App/Actions/Favorites/PinItemAction.cs [hunk 1]
```diff
@@ -3,53 +3,82 @@
...
 		{
-			context.PropertyChanged += Context_PropertyChanged;
+			if (context.HasSelection)
+			{
+				var items = context.SelectedItems.Select(x => x.ItemPath).ToArray();
+				await service.PinToSidebar(items);
+			}
+			else if (context.Folder is not null)
+			{
+				await service.PinToSidebar(context.Folder.ItemPath);
```
Comment (hecksmosis) [inline src/Files.App/Actions/Favorites/PinItemAction.cs:45]: I think we should remove the curly braces

File: src/Files.App/Actions/Favorites/UnpinItemAction.cs [hunk 2]
```diff
@@ -3,57 +3,80 @@
...
 		{
-			context.PropertyChanged += Context_PropertyChanged;
+			if (context.HasSelection)
+			{
+				var items = context.SelectedItems.Select(x => x.ItemPath).ToArray();
+				await service.UnpinFromSidebar(items);
+			}
+			else if (context.Folder is not null)
+			{
+				await service.UnpinFromSidebar(context.Folder.ItemPath);
```
Comment (hecksmosis) [inline src/Files.App/Actions/Favorites/UnpinItemAction.cs:44]: As above

File: src/Files.App/Actions/Favorites/PinItemAction.cs (hunk 1 above)
Comment (cinqmilleans) [reply src/Files.App/Actions/Favorites/PinItemAction.cs:44]: It's done.

File: src/Files.App/Actions/Favorites/UnpinItemAction.cs (hunk 2 above)
Comment (cinqmilleans) [reply src/Files.App/Actions/Favorites/UnpinItemAction.cs:43]: It's done.

File: src/Files.App/Actions/Favorites/PinItemAction.cs (hunk 1 above)
Comment (yaira2) [reply src/Files.App/Actions/Favorites/PinItemAction.cs:44]: We usually use curly braces if there is an else statement.

Comment (hecksmosis) [reply src/Files.App/Actions/Favorites/PinItemAction.cs:44]: Oh, I wasn't aware of that

File: src/Files.App/Actions/Favorites/UnpinItemAction.cs (hunk 2 above)
Comment (cinqmilleans) [reply src/Files.App/Actions/Favorites/UnpinItemAction.cs:44]: I put the style back as before.

File: src/Files.App/Actions/Favorites/PinItemAction.cs (hunk 1 above)
Comment (cinqmilleans) [reply src/Files.App/Actions/Favorites/PinItemAction.cs:45]: I put the style back as before.

**Review by yaira2 — APPROVED** [review]
LGTM

**Pull request merged**